src/main/java/edu/cornell/kfs/paymentworks/service/impl/PaymentWorksNewVendorConversionServiceImpl.java

```diff
@@ -289,92 +291,143 @@ protected void setPOAddressOnPaymentWorksVendor(VendorDetail vendorDetail, Payme
+            paymentWorksNewVendor.setBankAcctNameOnAccount(paymentWorksBankAccountDTO.getName_on_acct());
+
+            populateNewVendorBankAddressAttributes(paymentWorksNewVendor, paymentWorksBankAccountDTO);
+        }
+	}
 
-		paymentWorksNewVendor.setVendorRequestId(paymentWorksNewVendorDetailDTO.getId());
+	private void populateNewVendorBankAddressAttributes(PaymentWorksVendor paymentWorksNewVendor, PaymentWorksBankAccountDTO paymentWorksBankAccountDTO) {
+        if (bankAddressExists(paymentWorksBankAccountDTO)) {
+            PaymentWorksBankAddressDTO paymentWorksBankAddressDTO = paymentWorksBankAccountDTO.getBank_address();
+
+            paymentWorksNewVendor.setBankAddressId(paymentWorksBankAddressDTO.getId());
+            paymentWorksNewVendor.setBankAddressName(paymentWorksBankAddressDTO.getName());
+            paymentWorksNewVendor.setBankAddressStreet1(paymentWorksBankAddressDTO.getStreet1());
+            paymentWorksNewVendor.setBankAddressStreet2(paymentWorksBankAddressDTO.getStreet2());
+            paymentWorksNewVendor.setBankAddressCity(paymentWorksBankAddressDTO.getCity());
+            paymentWorksNewVendor.setBankAddressState(paymentWorksBankAddressDTO.getState());
+            paymentWorksNewVendor.setBankAddressCountry(paymentWorksBankAddressDTO.getCountry());
+            paymentWorksNewVendor.setBankAddressZipCode(paymentWorksBankAddressDTO.getZipcode());
+            paymentWorksNewVendor.setBankAddressValidated(paymentWorksBankAddressDTO.getValidated());
+        }
+	}
 
-		return paymentWorksNewVendor;
+	private void populateNewVendorCorporateAddressAttributes(PaymentWorksVendor paymentWorksNewVendor, PaymentWorksNewVendorDetailDTO paymentWorksNewVendorDetailDTO) {
+        if (corporateAddressExists(paymentWorksNewVendorDetailDTO)) {
+            PaymentWorksCorpAddressDTO corpAddress = paymentWorksNewVendorDetailDTO.getRequesting_company().getCorp_address();
+
+            paymentWorksNewVendor.setCorpAddressId(corpAddress.getId());
+            paymentWorksNewVendor.setCorpAddressName(corpAddress.getName());
+            paymentWorksNewVendor.setCorpAddressStreet1(corpAddress.getStreet1());
+            paymentWorksNewVendor.setCorpAddressStreet2(corpAddress.getStreet2());
+            paymentWorksNewVendor.setCorpAddressCity(corpAddress.getCity());
+            paymentWorksNewVendor.setCorpAddressState(corpAddress.getState());
+            paymentWorksNewVendor.setCorpAddressCountry(corpAddress.getCountry());
+            paymentWorksNewVendor.setCorpAddressZipCode(corpAddress.getZipcode());
+            paymentWorksNewVendor.setCorpAddressValidated(corpAddress.getValidated());
+        }
 	}
 
+    private boolean singleRemittanceAddressExists(PaymentWorksNewVendorDetailDTO paymentWorksNewVendorDetailDTO) {
+        if (ObjectUtils.isNotNull(paymentWorksNewVendorDetailDTO.getRequesting_company().getRemittance_addresses()) &&
```

Comment:
I would just return the boolean expression here.  Also, do you want to use StringUtills.isNotEmpty?  Similar thoughts for the other "exists" functions below

Reply:
I do not think that StringUtills is correct as these "exists" functions are looking for the existence of complex data structures which I think would be Objects (i.e. lists comprised of other class DTOs --OR-- class DTO's comprised of other multiple class DTOs).

Changing to return just the boolean and removing the if-then-else statements.